pkg/runner/job_executor.go

```diff
@@ -100,7 +100,15 @@ func newJobExecutor(info jobInfo, sf stepFactory, rc *RunContext) common.Executo
 	pipeline = append(pipeline, steps...)
 
 	return common.NewPipelineExecutor(pipeline...).
-		Finally(postExecutor).
+		Finally(func(ctx context.Context) error {
+			activeContext := ctx
+			if activeContext.Err() == context.Canceled {
+				// in case of an aborted run, we still should execute the
+				// post steps to allow cleanup.
+				activeContext = context.Background()
```

Comment:
For a single `SIGINT` I would expect that act executes steps and **jobs**  with `if: cancelled()` or similar and the post steps.

> ... doesn't want anything to run

Fair point, but then I don't understand why `SIGTERM` would execute post steps. The container is destroyed after rerunning act, if you don't use `--reuse`. I'm not shure why post steps should be more important than a step on `cancelled()`. It seems like that this impl can cause a deadlock, because you create a new context without any timeout.

Reply:
The post jobs could be more than stopping processes. They are meant as cleanup.
An example could be a note to a Webservice about an aborted build.
Indeed you are right with the canceled jobs (and post-if by default is `always` which should be executed as well).

Reply:
> It seems like that this impl can cause a deadlock, because you create a new context without any timeout.

Can you explain where you see the cause of a deadlock here?
A context by itself will not create deadlocks. In combination with go routines this might be the case but I don't create one here.

Reply:
Isn't this more a different issue? `act` currently has no such timeout, right?

Reply:
It is a different issue, because exec was not cancellable before this PR..., but you are changeing more than that.

Reply:
Hmm, I think I really don't see your point here.
- it is a different issue
- act was not being able to be canceled (it could have stuck before)
- it could get stuck now

Why should we change it in here? We should put that into a different PR, shouldn't we?

> but you are changeing more than that

What do you mean with that?

Reply:
Ah now I understand. Yes you are right. Sorry for maybe arguing too aggressive. :smiley: 

I will allow the cleanup context to be canceled after 5 minutes. That's a good idea.

Reply:
@ChristopherHX done